leetcode/src/5.c

```diff
@@ -0,0 +1,34 @@
+char * longestPalindrome(char * s){
+	char *result = NULL;
+	int strSize = 0;
+	for(int i = 0; *(s+i) != '\0'; i++){
+		strSize = i + 1;
+	}
+	int start = 0;
+	int end = 0;
+	int largest = 0;
+	for(int i = 0; i < strSize; i++){
+		for(int j = i + largest; j < strSize; j++){
+			if(*(s+i) == *(s+j)){
+				int pass = 1;
+				for(int k = 1; k <= (j-i)/2; k++){
```

Comment:
It is nice when you can split it by another function (ex: `bool checkPalindrome(char *s, int start, int end)`)?

Reply:
I mean:
```c

int checkPalindrome(char *s, int start, int end) {
    for(int k = 1; k <= (end-start)/2; k++) {
        if(*(s + start + k) != *(s + end - k))
            return 0;
    }
    return 1;
}
char * longestPalindrome(char * s){
	char *result = NULL;
	int strSize = strlen(s);
	int start = 0;
	int end = 0;
	int largest = 0;
	for(int i = 0; i < strSize; i++){
		for(int j = i + largest; j < strSize; j++){
			if(*(s+i) == *(s+j)){
				int pass = checkPalindrome(s, i, j);
				if(pass && largest < j-i+1){
					largest = j - i + 1;
					start = i;
					end = j;
				}
			}
		}
	}
	result = (s+start);
	// Destructive
	if(largest > 0 && end+1 != strSize+1){
		s[end+1] = '\0';
	}
	return result;
}
```

Reply:
Thanks, I wasn't sure if they were meant to be standalone functions or not